### ISSUE TITLE
table_exists? should not return true for views

### ISSUE THREAD

**Issue by yahonda** [rsim/oracle-enhanced#1179]
### Environment:

Rails master branch
Oracle enhanced adapter master branch
Oracle Database 12.1.0.2 Enterprise Edition
ruby 2.5.0dev (2017-02-13 trunk 57613) [x86_64-linux]

### Steps to reproduce:

```ruby
$ ARCONN=oracle bundle exec ruby -w -Itest test/cases/view_test.rb -n /test_table_exists/
... snip ...
Finished in 0.500429s, 3.9966 runs/s, 3.9966 assertions/s.

  1) Failure:
ViewWithPrimaryKeyTest#test_table_exists [test/cases/view_test.rb:48]:
'ebooks'' table should not exist


  2) Failure:
ViewWithoutPrimaryKeyTest#test_table_exists [test/cases/view_test.rb:133]:
'paperbacks' table should not exist

2 runs, 2 assertions, 2 failures, 0 errors, 0 skips
$
```

Right now, `table_exists?` just calls `data_source_exists?` which returns tables and views also considering database objects via database links. I'm thinking about to implement `table_exists?` just returns tables within the connected schema in Rails 5.1+.

**Comment by yahonda** [comment]
Working on it at https://github.com/yahonda/oracle-enhanced/tree/table_exists_warnings branch.

There are two regressions about handling quote in table names.

```ruby
$ ARCONN=oracle bundle exec ruby -w -Itest test/cases/view_test.rb -n test_table_exists
... snip ...

# Running:

E.

Finished in 0.458737s, 4.3598 runs/s, 2.1799 assertions/s.

  1) Error:
ViewWithPrimaryKeyTest#test_table_exists:
ActiveRecord::StatementInvalid: OCIError: ORA-01756: quoted string not properly terminated:           SELECT owner, table_name
          FROM all_tables
          WHERE owner = 'ARUNIT'
          AND table_name = 'ebooks''

    stmt.c:82:in oci8lib_230.so
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/cursor.rb:28:in `initialize'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/oci8.rb:175:in `new'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/oci8.rb:175:in `parse_internal'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/oci8.rb:168:in `parse'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `block in delegating_block'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced/oci_connection.rb:109:in `prepare'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced/database_statements.rb:26:in `block in exec_query'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:611:in `block (2 levels) in log'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:610:in `block in log'
    /home/yahonda/git/rails/activesupport/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:602:in `log'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb:1059:in `log'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced/database_statements.rb:22:in `exec_query'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:361:in `select'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:42:in `select_all'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb:97:in `select_all'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:68:in `select_rows'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:62:in `select_values'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb:605:in `table_exists?'
    test/cases/view_test.rb:48:in `test_table_exists'

2 runs, 1 assertions, 0 failures, 1 errors, 0 skips
$
```

```ruby
$  ARCONN=oracle bundle exec ruby -w -Itest test/cases/adapter_test.rb -n test_table_exists?
... snip ...
# Running:

E

Finished in 0.026382s, 37.9041 runs/s, 113.7124 assertions/s.

  1) Error:
ActiveRecord::AdapterTest#test_table_exists?:
ActiveRecord::StatementInvalid: OCIError: ORA-01756: quoted string not properly terminated:           SELECT owner, table_name
          FROM all_tables
          WHERE owner = 'ARUNIT'
          AND table_name = '''

    stmt.c:82:in oci8lib_230.so
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/cursor.rb:28:in `initialize'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/oci8.rb:175:in `new'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/oci8.rb:175:in `parse_internal'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/ruby-oci8-2.2.3/lib/oci8/oci8.rb:168:in `parse'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `block in delegating_block'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced/oci_connection.rb:109:in `prepare'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced/database_statements.rb:26:in `block in exec_query'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:611:in `block (2 levels) in log'
    /home/yahonda/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:610:in `block in log'
    /home/yahonda/git/rails/activesupport/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:602:in `log'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb:1059:in `log'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced/database_statements.rb:22:in `exec_query'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:361:in `select'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:42:in `select_all'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb:97:in `select_all'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:68:in `select_rows'
    /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:62:in `select_values'
    /home/yahonda/git/oracle-enhanced/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb:605:in `table_exists?'
    test/cases/adapter_test.rb:45:in `test_table_exists?'

1 runs, 3 assertions, 0 failures, 1 errors, 0 skips
$
```

**Comment by yahonda** [comment]
Fixed by #1191
